src/main/drivers/barometer/barometer_bmp388.c

```diff
@@ -0,0 +1,414 @@
+#define BMP388_INT_RESERVED_5_BIT           5
+#define BMP388_INT_DRDY_EN_BIT              6
+#define BMP388_INT_RESERVED_7_BIT           7
+
+// OSR register
+#define BMP388_OSR_P_BIT                    0 // to 2
+#define BMP388_OSR4_T_BIT                   3 // to 5
+#define BMP388_OSR_P_MASK                   (0x03)  // -----111
+#define BMP388_OSR4_T_MASK                  (0x38)  // --111---
+
+// configure pressure and temperature oversampling, forced sampling mode
+#define BMP388_PRESSURE_OSR              (BMP388_OVERSAMP_8X)
+#define BMP388_TEMPERATURE_OSR           (BMP388_OVERSAMP_1X)
+
+// see Datasheet 3.11.1 Memory Map Trimming Coefficients
+typedef struct bmp388_calib_param_s {
+    uint16_t T1;
+    uint16_t T2;
+    int8_t T3;
+    int16_t P1;
+    int16_t P2;
+    int8_t P3;
+    int8_t P4;
+    uint16_t P5;
+    uint16_t P6;
+    int8_t P7;
+    int8_t P8;
+    int16_t P9;
+    int8_t P10;
+    int8_t P11;
+} __attribute__((packed)) bmp388_calib_param_t;
+
+STATIC_ASSERT(sizeof(bmp388_calib_param_t) == BMP388_TRIMMING_DATA_LENGTH, bmp388_calibration_structure_incorrectly_packed);
+
+static uint8_t bmp388_chip_id = 0;
+STATIC_UNIT_TESTED bmp388_calib_param_t bmp388_cal;
+// uncompensated pressure and temperature
+uint32_t bmp388_up = 0;
+uint32_t bmp388_ut = 0;
+STATIC_UNIT_TESTED int64_t t_lin = 0;
+
+static void bmp388_start_ut(baroDev_t *baro);
+static void bmp388_get_ut(baroDev_t *baro);
+static void bmp388_start_up(baroDev_t *baro);
+static void bmp388_get_up(baroDev_t *baro);
+
+STATIC_UNIT_TESTED void bmp388_calculate(int32_t *pressure, int32_t *temperature);
+
+void bmp388_extiHandler(extiCallbackRec_t* cb)
+{
+#ifdef DEBUG
+    static uint32_t bmp388ExtiCallbackCounter = 0;
+
+    bmp388ExtiCallbackCounter++;
+#endif
+
+    baroDev_t *baro = container_of(cb, baroDev_t, exti);
+
+    uint8_t intStatus = 0;
+    busReadRegisterBuffer(&baro->busdev, BMP388_INT_STATUS_REG, &intStatus, 1);
+}
+
+void bmp388BusInit(busDevice_t *busdev)
+{
+#ifdef USE_BARO_SPI_BMP388
+    if (busdev->bustype == BUSTYPE_SPI) {
+        IOHi(busdev->busdev_u.spi.csnPin); // Disable
+        IOInit(busdev->busdev_u.spi.csnPin, OWNER_BARO_CS, 0);
+        IOConfigGPIO(busdev->busdev_u.spi.csnPin, IOCFG_OUT_PP);
+        spiBusSetDivisor(busdev, SPI_CLOCK_STANDARD);
+    }
+#else
+    UNUSED(busdev);
+#endif
+}
+
+void bmp388BusDeinit(busDevice_t *busdev)
+{
+#ifdef USE_BARO_SPI_BMP388
+    if (busdev->bustype == BUSTYPE_SPI) {
+        spiPreinitByIO(busdev->busdev_u.spi.csnPin);
+    }
+#else
+    UNUSED(busdev);
+#endif
+}
+
+void bmp388BeginForcedMeasurement(busDevice_t *busdev)
+{
+    // enable pressure measurement, temperature measurement, set power mode and start sampling
+    uint8_t mode = (BMP388_MODE_FORCED << 4) | (1 << 1) | (1 << 0);
+    busWriteRegister(busdev, BMP388_PWR_CTRL_REG, mode);
+}
+
+bool bmp388Detect(const bmp388Config_t *config, baroDev_t *baro)
+{
+    delay(20);
+
+#if defined(BARO_EXTI_PIN)
```

Comment:
This should not be dependent on `BARO_EXTI_PIN` being defined, as Unified Targets can define a pin at runtime.

Reply:
ok, yes good spot.  much of this code pre-dates some of the more recent unfied target changes and needs updating a little. I will do it.

Reply:
done.